src/app/add-new-todo-list/add-new-todo-list.component.html

```diff
@@ -0,0 +1,6 @@
+<h1>{{title}}</h1>
+<input type='text' placeholder='Add new todo item'>
+<div *ngFor='let todo of data.todos'>
+  {{todos.title}}
+  </div>
+<app-add-new-todo-list></app-add-new-todo-list>
```

Comment:
This seems a recursive instance of the component you are defining.
Remove this line and add it to another component (e.g. app.component)